src/makeInfinityAdder.js

```diff
@@ -1,12 +1,26 @@
 'use strict';
 
 /***
- *
  * @return {function}
  */
-
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(num) {
+    if (num === undefined) {
+      const result = sum;
+
+      sum = 0; // reset sum after returning the result
+
+      return result;
+    } else {
+      sum += num;
```

Comment:
Ensure that the `num` parameter is a number before adding it to `sum`. You might want to add a check to handle non-numeric inputs gracefully.